### PR TITLE
Refresh stale forms after 30 min of no change

### PR DESCRIPTION
**Pull request by nmajor25**
Shortcut Story ID: [sc-50208]

- [x] Add tests

### CODE REVIEW

File: src/js/services/forms.js
Comment (nmajor25) [inline src/js/services/forms.js:1]: When the form initially loads or reloads (on `ready:form`), set the `refreshForm()` function to run in 30 min.

When the user makes a change to the form (represented by the `updateStoredSubmission()` function being called), set a new `refreshForm()` to be called in 30 min.

When the user submits the form, cancel the scheduled `refreshForm()` function.

File: src/js/services/forms.js [hunk 1]
```diff
@@ -48,6 +51,8 @@ export default App.extend({
   },
   readyForm() {
     this.trigger('ready');
+
+    this.refreshForm();
```
Comment (nmajor25) [inline src/js/services/forms.js:55]: I initially put this in the `initialize()` function. But if a user is prompted with the keep/discard work prompt, `this.refreshForm()` would be called before the actual form has been shown to the user.

So I put it here instead. To wait until the form content is loaded.

**Review by paulfalgout — APPROVED** [review]
Looks correct to me

**Comment by nmajor25** [comment]
@paulfalgout cool, I'll get some tests created for this.

File: test/integration/forms/action.js
Comment (nmajor25) [inline test/integration/forms/action.js:1]: The test is designed to simulate a user having a form open in their browser that becomes "stale". And after 30 minutes of inactivity, we reload the form so the current state of the form is displayed to the user.

Test sequence:

1. Load an empty form.
2. Restart the form after 30 minutes of no changes (via `cy.tick()`). The form is reloaded with a new draft latest form response.
    - Simulating a form draft having been created in a different tab/window/session.
4. Verify that the form loaded the latest form response data into the form. 
5. Restart the form after 30 more minutes of no changes (via `cy.tick()`). The form is reloaded with a new submitted form response.
    - Simulating a form draft having been submitted in a different tab/window/session.
6. Verify that the form is now submitted. Content shown should be from the submitted form response and the submit button should be in the `Update` state.

**Pull request merged**